sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/EventHubSharedKeyCredential.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Messaging.EventHubs.Core;
+
+namespace Azure.Messaging.EventHubs.Authorization
+{
+    /// <summary>
+    ///   Provides a credential based on a shared access signature for a given
+    ///   Event Hub instance.
+    /// </summary>
+    ///
+    /// <seealso cref="Azure.Core.TokenCredential" />
+    ///
+    public sealed class EventHubSharedKeyCredential : TokenCredential
```

Comment:
I'm surprised that for once, Java is more verbose than C#! ours is EventHubSharedAccessKeyCredential. 🤣 Gotta get all those words in there.